PostFreeSurfer/scripts/CreateMyelinMaps.sh

```diff
@@ -242,60 +237,125 @@ for STRING in "$AtlasSpaceFolder"/"$NativeFolder"@native@roi "$AtlasSpaceFolder"
+	${AtlasSpaceFolder}/${NativeFolder}/${Subject}.L.sphere.MSMSulc.native.surf.gii \
+	${AtlasSpaceFolder}/${Subject}.L.sphere.${HighResMesh}k_fs_LR.surf.gii \
+	BARYCENTRIC ${StudyFolder}/${Subject}/T1w/${Subject}.L.midthickness.${HighResMesh}k_fs_LR.surf.gii
+${CARET7DIR}/wb_command -surface-resample ${StudyFolder}/${Subject}/T1w/${NativeFolder}/${Subject}.R.midthickness.native.surf.gii \
+	${AtlasSpaceFolder}/${NativeFolder}/${Subject}.R.sphere.MSMSulc.native.surf.gii \
+	${AtlasSpaceFolder}/${Subject}.R.sphere.${HighResMesh}k_fs_LR.surf.gii \
+	BARYCENTRIC ${StudyFolder}/${Subject}/T1w/${Subject}.R.midthickness.${HighResMesh}k_fs_LR.surf.gii
 		
-		# Myelin Map BC using low res
-		"$HCPPIPEDIR"/global/scripts/MyelinMap_BC.sh \
-			--study-folder="$StudyFolder" \
-			--subject="$Subject" \
-			--registration-name="MSMSulc" \
-			--use-ind-mean="$UseIndMean" \
-			--low-res-mesh="$LowResMesh" \
-			--myelin-target-file="$MyelinTargetFile" \
-			--map="$MyelinMap"
-		# ----- End moved statements -----
-		# bias field is computed in the module MyelinMap_BC.sh
-		BiasFieldComputed=true
-		${CARET7DIR}/wb_command -cifti-separate ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.BiasField.native.dscalar.nii COLUMN \
-			-metric CORTEX_LEFT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.L.BiasField.native.func.gii \
-			-metric CORTEX_RIGHT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.R.BiasField.native.func.gii
-	else
-		# bias field in native space is already generated
-		# BC the other types of given myelin maps
-		${CARET7DIR}/wb_command -cifti-math "Var - Bias" ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.${MyelinMap}_BC.native.dscalar.nii -var Var ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.${MyelinMap}.native.dscalar.nii -var Bias ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.BiasField.native.dscalar.nii
+# BC processing
+if [ "${T2wPresent}" = "YES" ] ; then	
+	# determine the resolution of the reference myelin map
+	# TODO: is there a better way to directly get the information of 164k, 59k or 32k? 1) hack into the xml, 2) hack info file names, 3)???
+	NumRefSurfVertice=$(${CARET7DIR}/wb_command -file-information "$ReferenceMyelinMaps" -only-cifti-xml | grep -m 1 -oP 'SurfaceNumberOf(Vertices|Nodes)="\K\d+')
+	# resample the reference map into 32k
+	case "$NumRefSurfVertice" in
+		("163842")
+			RefResMesh="164"
+			SphereFolder=${AtlasSpaceFolder}
+			T1wSurfFolder=${StudyFolder}/${Subject}/T1w
+			;;
+		("59292")
+			RefResMesh="59"
+			SphereFolder=${AtlasSpaceFolder}/fsaverage_LR59k
+			T1wSurfFolder=${StudyFolder}/${Subject}/T1w/fsaverage_LR59k
+			;;
+		("32492")
+			RefResMesh="32"
+			SphereFolder=${AtlasSpaceFolder}/fsaverage_LR32k
+			T1wSurfFolder=${StudyFolder}/${Subject}/T1w/fsaverage_LR32k
+			;;
+		(*)
+			log_Err_Abort "unrecognized NumRefSurfVertice value '$NumRefSurfVertice', valid options are 163842, 59292, 32492"
```

Comment:
We have additional options in NHPs.  We probably should test if equal to 163842 or else assume it matches one of the LowResMeshes.

Reply:
I don't get it. How to "assume it matched one of the $LowResMeshes"? I would suggest to list all the valid options in this case-esac statement to check if the reference myelin map falls into valid options or not.

Reply:
There could be a variety of numbers.  The possibilities are either the 164k mesh or one of the user specified LowResMeshes to the PostFreeSurfer pipeline.

Reply:
I see. I just feel it is technically difficult to implement without knowing the ground truth relation between the number of vertices and the corresponding LowResMesh.

Reply:
Maybe @coalsont  has a suggestion.

Reply:
One thing I would like to confirm is, except for the case `ResMesh="164" when NumRefSurfVertice=163842`, do all the rest cases have the relation `echo "$NumRefSurfVertice" | cut -c 1-2 == $ResMesh`?

Reply:
@takuya-hayashi might know

Reply:
Don't count on taking the first digits.  If anything, it would probably be rounding the `/ 1000` to the nearest integer, but they are not automatically computed (someone looked at the number, rounded it in their head, and used that for the filename of the spheres).

I don't see a reason why CreateMyelinMaps (and PostFreeSurfer itself) can't just expect the template myelin map to be HighResMesh.  This is something that we will rarely regenerate, and we can resample it to the appropriate mesh for the species at that time, so I see no substantial benefit to supporting other meshes in the per-subject code.  I also don't really like having the script modify what commands it runs based on the contents of a file, rather than based on options passed in by the user (which would suggest an option like `--myelin-template-mesh=32`, and logic that uses HighResMesh otherwise).

If we decide we really, truly don't want to have to resample the myelin template to the HighRes mesh in advance...does this line contain what you need to proceed?

https://github.com/Washington-University/HCPpipelines/blob/a8cea8aeee6bfc53dfd5e7936aad9bf6505e17c4/PostFreeSurfer/scripts/CreateMyelinMaps.sh#L214

Reply:
The issue is that we haven't been terribly consistent about these myelin templates (e.g., the original MSMSulc template was 164k.  The MSMAll template was 32k.  The transmit field corrected MSMAll template was 32k.  Macaques could be yet different resolutions).  Thus, we need some flexibility here if we want to maintain backwards compatibility, which I think we should.